### DISCUSSION TITLE
Recently hosted Renovate is unable to look at git@ tags

### DISCUSSION THREAD

**Discussion by MindTooth** [renovatebot/renovate#22557] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Links like these used to work (git-tags manager):
```
git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git
```

But are now throwing following instead:

>     "message": "Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n",

They stopped working two days ago.  Running `renovate --platform local` works just fine.

* Project: https://github.com/MindTooth/renovate-git-tags-test
* Dependency Dashboard w/ error: https://github.com/MindTooth/renovate-git-tags-test/issues/3

An excerpt of the log is below.  I can provide more if needed, but it used to work, now it don't, and the log is the only thing that's is different.  I've not changed anything with the projects over at GitLab.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Failed to look up git-tags package git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git
{
  "dependency": "git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git"
  "packageFile": "Puppetfile"
}

DEBUG: Datasource unknown error
{
  "datasource": "git-tags"
  "packageName": "git@gitlab.com:hitchhikers/puppet-modules/puppetlabs-stdlib.git"
  "err": {
    "task": {
      "commands": [
        "ls-remote",
        "git@gitlab.com:hitchhikers/puppet-modules/puppetlabs-stdlib.git"
      ],
      "format": "utf-8",
      "parser": "[function]"
    },
    "message": "Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n",
    "stack": "Error: Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n\n    at Object.action (/opt/buildpack/tools/renovate/35.105.2/node_modules/simple-git/src/lib/plugins/error-detection.plugin.ts:42:29)\n    at PluginStore.exec (/opt/buildpack/tools/renovate/35.105.2/node_modules/simple-git/src/lib/plugins/plugin-store.ts:29:29)\n    at /opt/buildpack/tools/renovate/35.105.2/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:127:42\n    at new Promise (<anonymous>)\n    at GitExecutorChain.handleTaskData (/opt/buildpack/tools/renovate/35.105.2/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:124:14)\n    at GitExecutorChain.<anonymous> (/opt/buildpack/tools/renovate/35.105.2/node_modules/simple-git/src/lib/runners/git-executor-chain.ts:100:40)\n    at Generator.next (<anonymous>)\n    at fulfilled (/opt/buildpack/tools/renovate/35.105.2/node_modules/simple-git/dist/cjs/index.js:55:24)"
  }
}
```

### Package:

```
          {
            "depName": "logrotate",
            "packageName": "git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git",
            "sourceUrl": "git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git",
            "gitRef": true,
            "currentValue": "v6.0.0",
            "datasource": "git-tags",
            "updates": [],
            "warnings": [
              {
                "topic": "git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git",
                "message": "Failed to look up git-tags package git@gitlab.com:hitchhikers/puppet-modules/voxpupuli-logrotate.git"
              }
            ],
            "versioning": "semver"
          },
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
I see the same error when running `yarn start --dry-run=lookup MindTooth/renovate-git-tags-test` locally from source, so it's not specific to the hosted application. I'm not sure why you see something different when running locally, perhaps because you use git@gitlab.com already on your local host for other things?

**Reply by rarkins** [reply]
I ran the following:

```
❯ ssh-keyscan -t rsa gitlab.com >> ~/.ssh/known_hosts
# gitlab.com:22 SSH-2.0-GitLab-SSHD
```

But it still didn't resolve the problem

**Reply by MindTooth** [reply]
You know what, this might have been a false negative.  Because, even logs as far back as two months ago had the same issue (18 Apr 2023 at 02:06:01).  And yes, one properly need a proper SSH setup for using Git over SSH.  So for it to be working inside my org could be for other factors.

Though, it begs the question, why did it "work" before and why it now changed all of the sudden. 🤔  

At least thanks for looking at it.  Usually an extra pair of eyes help in narrowing down the cause.

**Reply by MindTooth** [reply, marked as answer]
Not sure if I fixed it myself with a host rule, but now it works again. 🤷🏻‍♂️ Meaning it must have had access before at some point, but not anymore.

**Reply by rushilsrivastava** [reply]
@MindTooth What was the host rule you ended up using?

**Reply by MindTooth** [reply]
Just as following: https://github.com/MindTooth/renovate-puppetfile-test/blob/dc2e9133714dddbc1fbcc66bae6448ab4ef759f7/.github/renovate.json#L4

```json
  "hostRules": [
    {
      "matchHost": "https://gitlab.com/hitchhikers/puppet-modules",
      "encrypted": {
        "token": "<token>"
      }
    }
  ],
```

So nothing fancy really.  But, I can’t wrap my head around why it worked at some point without a hostRule. 🤷🏻‍♂️
